### PR TITLE
reject imported memories early

### PR DESCRIPTION
**Pull request by Ekleog-NEAR**
They were already rejected by wasmer2 before this change, just after preparation. So this should not be a protocol change.

Second attempt at #8029, without a protocol change this time

### CODE REVIEW

File: runtime/near-vm-runner/src/prepare.rs [hunk 1]
```diff
@@ -223,11 +221,10 @@ impl<'a> ContractModule<'a> {
...
             let type_idx = match *import.external() {
                 External::Function(ref type_idx) => type_idx,
-                External::Memory(ref memory_type) => {
-                    imported_mem_type = Some(memory_type);
-                    continue;
-                }
+                External::Memory(_) => return Err(PrepareError::Memory),
                 _ => continue,
+                // TODO: it’s a protocol change because the gas amount burned would change for some tests, but here we should
+                // _ => return Err(PrepareError::Instantiate),
```
Comment (nagisa) [inline runtime/near-vm-runner/src/prepare.rs:227]: Allowing imports of globals etc. seems fine to me, even if we don’t have any currently. As long as we confirm that they must come from `env` which we do, it is fine.

File: runtime/near-vm-runner/src/tests/runtime_errors.rs [hunk 2]
```diff
@@ -77,6 +77,30 @@ fn test_multiple_memories() {
         ]);
 }
 
+#[test]
+fn test_multiple_memories() {
+    test_builder()
+        //      | wasm header  | version 1     | mem sect   | 1st mem  | 2nd mem    |
+        //      |              |               | 2 elt 7 byt|          |            |
+        .wasm(b"\x00\x61\x73\x6d\x01\x00\x00\x00\x05\x07\x02\x01\x01\x02\x01\x03\x04")
```
Comment (nagisa) [inline runtime/near-vm-runner/src/tests/runtime_errors.rs:85]: Should `wat` be preferred instead? It should support encoding multiple memories just fine and should end up much more readable too?

File: runtime/near-vm-runner/src/prepare.rs (hunk 1 above)
Comment (Ekleog-NEAR) [reply runtime/near-vm-runner/src/prepare.rs]: Hmm so I was coming from the idea that right now we validate func types again (though I’m not really sure why), but we just ignore globals etc. Anyway it doesn’t change anything, so I’m removing this comment :)

File: runtime/near-vm-runner/src/tests/runtime_errors.rs (hunk 2 above)
Comment (Ekleog-NEAR) [reply runtime/near-vm-runner/src/tests/runtime_errors.rs]: Oh I expected it to fail encoding, but it does work! Much better, thanks :)

**Review by nagisa — APPROVED** [review]

**Pull request merged**